### ISSUE TITLE
study view: add chart with gene panels

### ISSUE THREAD

**Issue by tmazor** [cBioPortal/cbioportal#6762]
I'm inspired by all the new data being added to study view...

For studies with gene panels, we should automatically create a chart in study view with the gene panel for each sample. The GENIE studies have this, but we should provide it for all studies with gene panel data.

**Comment by jjgao** [comment]
@tmazor @n1zea144 @zhx828 @kalletlak : I think we have two options here:

1. create clinical attribute of gene panels automatically when importing gene panel data
2. create a new end point that pull from gene panel data and show pie charts

For option 1, one issue is that we may have different gene panels for different profiles, and therefore, the current `gene_panel` attribute may not work.

So maybe we should go with option 2? 

For option 2, maybe we can work on this after finish #6771? We can then display a gene_panel icon for the profiles with gene panels and the user can hover and show the gene panel pie chart per profile. They can also show it from other places such as Mutated Genes table.

**Comment by n1zea144** [comment]
I wonder how often we actually have different panels for different profiles.  Most target studies have cna/mutation data, but we never get cna panel information (its always assumed to be the same panel).

**Comment by jjgao** [comment]
@n1zea144 Good point. Maybe we only need one gene panel per sample. If that's the expectation, we should code it that way in database. 

I could think of fusion data on different panel and RPPA data on different panel, but they are currently not properly recorded currently.

**Comment by tmazor** [comment]
I agree, I think we need to allow for different gene panels for different profiles - while not super common this does happen (as an example, TCGA GBM 2008 has a gene panel for mutations but used a different assay for copy number analysis of all genes) and as @jjgao points out it is even more common for other datatypes like fusions.

**Comment by tmazor** [comment]
we should still do this

**Comment by jjgao** [comment]
Now that we have the `Genome Profile Sample Counts` table, maybe we can add an icon to a profile if gene panels belong to it? Clicking on the icon can bring up a piechart. While the pie chart will not be open by default, it'll be saved to the user profile once opened.

**Comment by jjgao** [comment]
@kalletlak I am wondering if you can help with this one when you have a chance?

**Comment by jjgao** [comment]
@Luke-Sikina I am wondering if you have the bandwidth to help with this one?

**Comment by tmazor** [comment]
@jjgao I like the idea to add an icon to the Genomic Profiles table to indicate which profiles are panel-based, but I'm not sure what the best way to add those charts is. Adding a new chart based on an icon in a table would be a totally new way to add a chart and I worry will be confusing or not obvious to users, since all other charts get added via the Charts menu.

What if we also have these gene panel charts in the Charts menu? We could still have the icon in the Genomic Profile table, and the icon could prompt users to the Charts menu? Although I guess that adds an extra step in adding the gene panel chart if the user is coming from the Genomic Profiles table, so maybe the gene panel charts can be added by both the Charts menu and by the Genomic Profiles table?

Also, should the table be called 'Molecular Profiles' rather than 'Genomic' ? And then we could also drop the column name and it would match the Cancer Type tables?
![image](https://user-images.githubusercontent.com/10508276/128876782-93d79228-ece8-4371-9969-e1f5eb8837bd.png)

**Comment by jjgao** [comment]
@tmazor Good point. I like the idea of having it in both Charts and Profiles table. It's good idea to keep it consistent so that one can add any chart from the Charts menu. It would also be useful to the have the icon in the Profiles table to indicate there are associated gene panels withe the profile and to provide a quick way to show the pie chat. For the profiles table, it may be better to just provide a toggle feature to turn on/off the pie chart instead of bringing our the bulky Charts menu. 

As for 'Genomic Profiles' vs 'Molecular Profiles', I am wondering if we should just go for 'Data Profiles'...

**Comment by tmazor** [comment]
Ooo yes, good point @jjgao  - we have already moved past just molecular data, so 'Data Profiles' sounds great to me!

And it sounds like we're in agreement in adding gene panel charts from both Charts menu and the Data Profiles table.
